internal/eva/settings.go

```diff
@@ -0,0 +1,123 @@
+package eva
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+const (
+	getSettingPath   = "/api/core/management/GetSetting"
+	setSettingPath   = "/api/core/management/SetSetting"
+	unsetSettingPath = "/api/core/management/UnsetSetting"
+)
+
+type SetSettingsRequest struct {
+	Key                string `json:"Key"`
+	Value              string `json:"Value"`
+	OrganizationUnitID int64  `json:"OrganizationUnitID,omitempty"`
+}
+
+type SetSettingsResponse struct {
+}
+
+func (c *Client) SetSettings(ctx context.Context, req SetSettingsRequest) (*SetSettingsResponse, error) {
+	resp, err := c.restClient.R().
+		SetBody(req).
+		Post(setSettingPath)
+
+	if err != nil {
+		tflog.Error(ctx, "An network error ocurred.", err)
+
+		return nil, err
+	}
+
+	if resp.StatusCode() != 200 {
```

Comment:
Does the EVA API always return status code `200` for every successful request? or could it also be `201` for creating a new resource or `204` for a deletion?

Reply:
Not full REST complaint, so sadly enough that is not the case.